Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Utils/Utils.h>
+#include <QMessageBox>
+#include <QVBoxLayout>
+
+#include "RobotImporter/RobotImporterWidget.h"
+#include "RobotImporter/RobotImporterWidgetUtils.h"
+#include "RobotImporter/URDF/RobotImporter.h"
+
+namespace ROS2
+{
+    namespace Internal
+    {
+        AZStd::string GetBaseName(AZStd::string path)
+        {
+            QFileInfo fileInfo(path.c_str());
+            return fileInfo.baseName().toStdString().c_str();
+        }
```

Comment:
There is already functionality in AzCore to do this, no need to re-create it here. Refer to Code/Framework/AzCore/AzCore/IO/Path/Path.h

```
...
class PathView 
{
...
    constexpr PathView Filename() const;
...

```
So the usage would look like this
```
AZ::IO::PathView(path).Filename().Native().c_str()
```


```